tests/Temporalio.Tests/Client/WorkflowHandleTests.cs

```diff
@@ -0,0 +1,126 @@
+namespace Temporalio.Tests.Client;
+
+using System;
+using Temporalio.Api.Enums.V1;
+using Xunit;
+using Xunit.Abstractions;
+
+public class WorkflowHandleTests : WorkflowEnvironmentTestBase
+{
+    public WorkflowHandleTests(ITestOutputHelper output, WorkflowEnvironment env)
+        : base(output, env)
+    {
+    }
+
+    [Fact]
+    public async Task SignalAsync_Simple_Succeeds()
+    {
+        // Start
+        var handle = await Client.StartWorkflowAsync(
+            IKitchenSinkWorkflow.Ref.RunAsync,
+            new KSWorkflowParams(ActionSignal: "SomeActionSignal"),
+            new(id: $"workflow-{Guid.NewGuid()}", taskQueue: Env.KitchenSinkWorkerTaskQueue));
+
+        // Send signal
+        await handle.SignalAsync(
+            IKitchenSinkWorkflow.Ref.SomeActionSignalAsync,
+            new KSAction(Result: new(Value: "Some String")));
+
+        // Confirm result
+        Assert.Equal("Some String", await handle.GetResultAsync());
+    }
+
+    [Fact]
+    public async Task QueryAsync_Simple_Succeeds()
+    {
+        // Start
+        var handle = await Client.StartWorkflowAsync(
+            IKitchenSinkWorkflow.Ref.RunAsync,
+            new KSWorkflowParams(new KSAction(QueryHandler: new(Name: "SomeQuery"))),
+            new(id: $"workflow-{Guid.NewGuid()}", taskQueue: Env.KitchenSinkWorkerTaskQueue));
+
+        // Issue query
+        var res = await handle.QueryAsync(IKitchenSinkWorkflow.Ref.SomeQuery, "Some String");
+
+        // Confirm result
+        Assert.Equal("Some String", res);
+    }
+
+    [Fact]
+    public async Task DescribeAsync_Simple_HasProperValues()
+    {
+        // Run
+        var handle = await Client.StartWorkflowAsync(
+            IKitchenSinkWorkflow.Ref.RunAsync,
+            new KSWorkflowParams(new KSAction(Result: new(Value: "Some String"))),
+            new(id: $"workflow-{Guid.NewGuid()}", taskQueue: Env.KitchenSinkWorkerTaskQueue));
+        Assert.Equal("Some String", await handle.GetResultAsync());
+
+        // Describe
+        var desc = await handle.DescribeAsync();
+
+        // Check values
+        var beforeNow = DateTime.UtcNow.AddSeconds(-30);
+        var afterNow = DateTime.UtcNow.AddSeconds(30);
+        var temp = desc.CloseTime!.Value;
+        Assert.InRange(desc.CloseTime!.Value, beforeNow, afterNow);
+        Assert.InRange(desc.ExecutionTime!.Value, beforeNow, afterNow);
```

Comment:
@cretz  Not sure if you want to add another dependency, but Fluent Assertions library is alot nicer to use and gives nicer output when assertions fail.

It has a helpers that let you test a date is `CloseTo` some time and you pass it a TimeSpan of how close
https://fluentassertions.com/datetimespans/ 

The object graph comparision also comes in handy
https://fluentassertions.com/objectgraphs/

Also useful is the ability to async retry something and still use assertions in the body (which gives nice errors when it fails)...
```
Func<Task> asyncRetry = async () =>
{
    var describeExecutionResponse = await _stepFunctions.DescribeExecutionAsync(new DescribeExecutionRequest
    {
        ExecutionArn = executionArn
    });
    describeExecutionResponse.Should().NotBeNull();
    describeExecutionResponse.Status.Should().Be(ExecutionStatus.SUCCEEDED);
};
await asyncRetry.Should().NotThrowAfterAsync(waitTime: 60.Seconds(), pollInterval: 2.Seconds());
```

Reply:
As the tests grow I very well may add that dependency or other test helpers (I don't mind adding dependencies to our test project). I haven't needed too much yet (and I have started an `AssertMore` package at the root of the project for a couple of them).